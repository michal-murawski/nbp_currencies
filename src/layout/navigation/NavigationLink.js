import React from 'react';
import { NavLink } from 'react-router-dom';
import { withStyles } from 'material-ui/styles';
import Button from 'material-ui/Button';
import PropTypes from 'prop-types';
import { NavigationLinkStyles } from './styles';

const NavigationLink = props => {
  const { classes, children, ...otherProps } = props;
  return (
    <Button>
      <NavLink className={classes.root} activeClassName={classes.active} {...otherProps}>
        {children}
      </NavLink>
    </Button>
  );
};

NavigationLink.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
    .isRequired,
};

export const NavigationLinkRaw = NavigationLink;

export default withStyles(NavigationLinkStyles)(NavigationLink);
