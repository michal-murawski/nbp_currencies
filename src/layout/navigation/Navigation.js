import React from 'react';
import PropTypes from 'prop-types';
import Toolbar from 'material-ui/Toolbar';
import NavigationLink from './NavigationLink';

const Navigation = props => {
  return (
    <Toolbar>
      <NavigationLink exact to="/">
        Currencies
      </NavigationLink>
      <NavigationLink to="/favourites">Favourites</NavigationLink>
    </Toolbar>
  );
};

Navigation.propTypes = {
  classes: PropTypes.object,
};

export default Navigation;
