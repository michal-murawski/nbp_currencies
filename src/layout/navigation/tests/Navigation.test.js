import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navigation from '../Navigation';
import NavigationLink from '../NavigationLink';

describe('layout/Navigation', () => {
  const wrapper = shallow(<Navigation />);

  it('should render properly the Navigation', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render NavigationLink', () => {
    expect(wrapper.find(NavigationLink)).toHaveLength(2);
  });
});
