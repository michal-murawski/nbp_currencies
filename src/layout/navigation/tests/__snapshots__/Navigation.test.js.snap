// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout/Navigation should render properly the Navigation 1`] = `
<WithStyles(Toolbar)>
  <WithStyles(NavigationLink)
    exact={true}
    to="/"
  >
    Currencies
  </WithStyles(NavigationLink)>
  <WithStyles(NavigationLink)
    to="/favourites"
  >
    Favourites
  </WithStyles(NavigationLink)>
</WithStyles(Toolbar)>
`;
